### PR TITLE
Couple of small fixes

### PR DESCRIPTION
**Pull request by migueldeicaza**
The first merely makes sure that we include the proper header file, otherwise this fails to compile with recent versions of Xcode.

The second patch fixes an issue that would corrupt the data stream when attempting to initialize the agent in non-blocking mode, as it is necessary to propagate the EAGAIN signal upstream when the transport returns EAGAIN.

### CODE REVIEW

File: src/channel.c [hunk 1]
```diff
@@ -1165,6 +1165,7 @@ static int channel_request_auth_agent(LIBSSH2_CHANNEL *channel,
         if(rc == LIBSSH2_ERROR_EAGAIN) {
             _libssh2_error(session, rc,
                            "Would block sending auth-agent request");
+	    return rc;
```
Comment (vszakats) [inline src/channel.c:1168]: This needs spaces for indent.

**Comment by vszakats** [comment]
Including `openssl/err.h` makes sense, but do you have an idea what does it have to do with Xcode? What is the error you're seeing?

File: src/channel.c (hunk 1 above)
Comment (migueldeicaza) [reply src/channel.c:1168]: Good call, I think my editor was going full tabs.

**Comment by vszakats** [comment]
`packet.c` patch: regression from #752.

**Comment by migueldeicaza** [comment]
Without it, a file mater on references an ERR_ symbol which has not been declared, and it becomes an error to call due to a warning-as-error for using an undeclared function.

**Comment by vszakats** [comment]
Got it, so that is a regression from #789. It seems that OpenSSL builds with `OPENSSL_NO_ENGINE` could have fallen into this after that patch.

**Pull request merged**